### PR TITLE
attempt to improve graceful server shutdown

### PR DESCRIPTION
**Pull request by wouterdb**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -256,7 +257,7 @@ def database_name():
                 params=[(True, True, False), (True, False, False), (False, True, False),
                         (False, False, False), (True, True, True)],
                 ids=["SSL and Auth", "SSL", "Auth", "Normal", "SSL and Auth with not self signed certificate"])
-async def server_multi(inmanta_config, postgres_db, database_name, request):
+async def server_multi(event_loop, inmanta_config, postgres_db, database_name, request):
```
Comment (arnaudsjs) [inline tests/conftest.py]: Why is the event_loop fixure required?

File: tests/conftest.py [hunk 2]
```diff
@@ -221,7 +221,7 @@ def database_name():
 
 
 @pytest.fixture(scope="function")
-async def server(inmanta_config, postgres_db, database_name):
+async def server(event_loop, inmanta_config, postgres_db, database_name):
```
Comment (arnaudsjs) [inline tests/conftest.py]: Why is the event_loop fixure required?

File: src/inmanta/server/agentmanager.py [hunk 3]
```diff
@@ -165,16 +165,14 @@ def get_agent_client(self, tid: uuid.UUID, endpoint: str) -> Optional[ReturnClie
 
     @gen.coroutine
     def start(self) -> NoneGen:
-        self.running = True
+        yield super().start()
```
Comment (arnaudsjs) [inline src/inmanta/server/agentmanager.py]: The running variable of AgentManager should be removed. It shields the running variable of the ServerSlice class.

File: src/inmanta/server/protocol.py [hunk 4]
```diff
@@ -381,6 +384,7 @@ def start(self) -> None:
         """
             Start the server slice.
         """
+        yield super().start()
```
Comment (arnaudsjs) [inline src/inmanta/server/protocol.py]: The prestart(), start() and stop() function of SessionManager have return type "None". Shouldn't it be NoneGen?

File: tests/conftest.py (hunk 1 above)
Comment (wouterdb) [reply tests/conftest.py]: to ensure proper ordering, not sure if it is really required, but can't hurt

**Pull request merged**